File: pyg_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/GoodsDescService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbGoodsDesc;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface GoodsDescService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbGoodsDesc> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    void add(TbGoodsDesc goodsDesc);


    /**
     * 修改
     */
    void update(TbGoodsDesc goodsDesc);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    TbGoodsDesc findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPage(TbGoodsDesc goodsDesc, int pageNum, int pageSize);

}

File: pyg_shop_web/src/main/webapp/js/service/goodsDescService.js
//服务层
app.service('goodsDescService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../goodsDesc/findAll');
    };
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../goodsDesc/findPage/' + page + '/' + rows);
    };
    //查询实体
    this.findOne = function (id) {
        return $http.get('../goodsDesc/findOne/' + id);
    };
    //增加
    this.add = function (entity) {
        return $http.post('../goodsDesc/add', entity);
    };
    //修改
    this.update = function (entity) {
        return $http.post('../goodsDesc/update', entity);
    };
    //删除
    this.dele = function (ids) {
        return $http.get('../goodsDesc/delete/' + ids);
    };
    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../goodsDesc/search/' + page + "/" + rows, searchEntity);
    }
});
